tezos/new_context/src/tezedge_context.rs

```diff
@@ -387,24 +432,31 @@ impl TezedgeIndex {
         path: &str,
         entry: &Entry,
         entries: &mut Vec<(ContextKeyOwned, ContextValue)>,
+        storage: &mut Storage,
     ) -> Result<(), MerkleError> {
         match entry {
-            Entry::Blob(blob) => {
+            Entry::Blob(blob_id) => {
                 // push key-value pair
-                entries.push((self.string_to_key(path), blob.clone()));
+                let blob = storage.get_blob(*blob_id)?;
+                entries.push((self.string_to_key(path), blob.to_vec()));
                 Ok(())
             }
             Entry::Tree(tree) => {
                 // Go through all descendants and gather errors. Remap error if there is a failure
```

Comment:
@sebastiencs lets remove this comment from here, I have no idea what it is about and when it was written, just makes things more confusing.